pkg/apis/pipeline/v1beta1/result_conversion.go

```diff
@@ -26,4 +43,21 @@ func (r *TaskResult) convertFrom(ctx context.Context, source v1.TaskResult) {
 		properties[k] = PropertySpec{Type: ParamType(v.Type)}
 	}
 	r.Properties = properties
+	if source.Default != nil {
+		r.Default = &ParamValue{
+			Type: ParamType(source.Default.Type),
+		}
+		switch source.Default.Type {
+		case v1.ParamTypeString:
+			r.Default.StringVal = source.Default.StringVal
+		case v1.ParamTypeArray:
+			r.Default.ArrayVal = source.Default.ArrayVal
+		default:
+			defaults := make(map[string]string)
+			for k, v := range source.Default.ObjectVal {
+				defaults[k] = v
+			}
+			r.Default.ObjectVal = defaults
+		}
+	}
```

Comment:
Seems to be the same logic as in lines 18:34. May be wrap it into a function and call it here?